File: spring-cloud-stream/kafka-consumer/src/main/java/com/livk/consumer/KafkaConsumer.java
package com.livk.consumer;

import com.livk.commons.SpringLauncher;
import com.livk.stream.entity.StreamMessage;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import reactor.core.publisher.Flux;

import java.util.function.Consumer;

/**
 * <p>
 * KafkaConsumer
 * </p>
 *
 * @author livk
 * @date 2022/2/15
 */
@Slf4j
@SpringBootApplication
public class KafkaConsumer {

    public static void main(String[] args) {
        SpringLauncher.run(args);
    }

    @Bean
    public Consumer<Flux<StreamMessage<String>>> send() {
        return kafkaMessageFlux -> kafkaMessageFlux.subscribe(kafkaMessage -> log.info("[{}]", kafkaMessage));
    }

}

File: spring-cloud-stream/kafka-consumer/src/main/kotlin/com/livk/consumer/KafkaConsumer.kt
package com.livk.consumer

import com.livk.stream.entity.StreamMessage
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean
import reactor.core.publisher.Flux
import java.util.function.Consumer

/**
 * @author livk
 */
@SpringBootApplication
open class KafkaConsumer {
    @Bean
    open fun send(): Consumer<Flux<StreamMessage<String>>> = Consumer { kafkaMessageFlux ->
        kafkaMessageFlux.subscribe { kafkaMessage ->
            log.info(
                "[{}]",
                kafkaMessage
            )
        }
    }

    companion object {
        private val log: Logger = LoggerFactory.getLogger(KafkaConsumer::class.java)
    }
}

fun main(args: Array<String>) {
    runApplication<KafkaConsumer>(*args)
}
